include/quic/stream.hpp

```diff
@@ -95,22 +94,159 @@ namespace oxen::quic
 
         void set_user_data(std::shared_ptr<void> data);
 
-        void send(bstring_view data, std::any keep_alive);
-
-        inline void send(bstring_view data) { send(data, std::move(data)); }
+        void send(bstring_view data, std::shared_ptr<void> keep_alive = nullptr);
 
         template <
                 typename CharType,
                 std::enable_if_t<sizeof(CharType) == 1 && !std::is_same_v<CharType, std::byte>, int> = 0>
-        void send(std::basic_string_view<CharType> data, std::any keep_alive)
+        void send(std::basic_string_view<CharType> data, std::shared_ptr<void> keep_alive = nullptr)
         {
             send(convert_sv<std::byte>(data), std::move(keep_alive));
         }
 
+        template <typename CharType>
+        void send(std::basic_string<CharType>&& data)
+        {
+            auto keep_alive = std::make_shared<std::basic_string<CharType>>(std::move(data));
+            std::basic_string_view<CharType> view{*keep_alive};
+            send(view, std::move(keep_alive));
+        }
+
         template <typename Char, std::enable_if_t<sizeof(Char) == 1, int> = 0>
         void send(std::vector<Char>&& buf)
         {
-            send(std::basic_string_view<Char>{buf.data(), buf.size()}, std::move(buf));
+            send(std::basic_string_view<Char>{buf.data(), buf.size()}, std::make_shared<std::vector<Char>>(std::move(buf)));
+        }
+
+      private:
+        // Implementations classes for send_chunks()
+
+        // chunk_sender: When sending chunks we construct *one* of these, then share its ownership
+        // across all the chunks in flight.  When each individual chunk gets destroyed, it called
+        // back into this to queue the next chunk, which this class then sends into the stream.
+        //
+        // Container -- can be a value or a pointer, but not a reference.
+        template <typename Container>
+        struct chunk_sender : std::enable_shared_from_this<chunk_sender<Container>>
+        {
+            static_assert(!std::is_reference_v<Container>, "chunk_sender requires a value or pointer, not a reference");
```

Comment:
throw?

Reply:
When you can `static_assert` (i.e. can check conditions at compile time) that's always a better solution to fail to compile when preconditions aren't met.